### PR TITLE
feat[cartesian]: Setting extra compile args and DaCe block size via env variables

### PR DESCRIPTION
**Pull request by stubbiali**
## Description

This PR adds the ability to read extra compile arguments and the DaCe block size from the environment. When the `dace:gpu` backend is used, the block size is also included in the stencil fingerprint. These features are quite useful for the PMAP model.

## Requirements

- [ ] All fixes and/or new features come with corresponding tests.
- [ ] Important design decisions have been documented in the approriate ADR inside the [docs/development/ADRs/](docs/development/ADRs/Index.md) folder.

If this PR contains code authored by new contributors please make sure:

- [ ] All the authors are covered by a valid contributor assignment agreement provided to ETH Zurich and signed by the employer if needed.
- [ ] The PR contains an updated version of the `AUTHORS.md` file adding the names of all the new contributors.

### CODE REVIEW

File: src/gt4py/cartesian/backend/dace_backend.py
Comment (FlorianDeconinck) [inline src/gt4py/cartesian/backend/dace_backend.py:546]:
```suggestion
        # To prevent conflict with 3rd party usage of DaCe config always make sure that any changes
        #  be under the temporary_config manager
        with dace.config.temporary_config():
```

File: src/gt4py/cartesian/config.py
Comment (FlorianDeconinck) [inline src/gt4py/cartesian/config.py:88]: That sounds a bit low. Unless we care about old hardware we can easily bump it to `64,8,1`

**Review by FlorianDeconinck — APPROVED** [review]
LGTM.

Couple of comments to verbose the `temp` usage which is critical for external use and maybe a more aggressive default.

File: src/gt4py/cartesian/config.py
Comment (FlorianDeconinck) [reply src/gt4py/cartesian/config.py]: Indeed it is, but since we have the liberty to reset a default, we can update it to something more likely. A part of me would like the system to crash if you are using an antic GPU ;)

Comment (stubbiali) [reply src/gt4py/cartesian/config.py]: Both parts make sense ;)

**Comment by havogt** [comment]
cscs-ci run

**Comment by stubbiali** [comment]
@FlorianDeconinck. I cannot merge this myself as I don't have write access, so could you please merge. Thanks a lot! 🙏🏻

**Pull request merged**
